interface/main/finder/multi_patients_finder.php

```diff
@@ -0,0 +1,258 @@
+$('#by-id, #by-name').select2({
+    ajax: {
+        url: 'multi_patients_finder_ajax.php',
+        data:function (params) {
+            var query = {
+                search: params.term,
+                type: $(this).attr('id')
+            }
+            return query;
+        },
+        dataType: 'json'
+    }
+});
+
+//get all the data of selected patient
+$('#by-id').on('change', function () {
+    $.ajax({
+        url: 'multi_patients_finder_ajax.php',
+        data:{
+            type:'patient-by-id',
+            search:$('#by-id').val()
+        },
+        dataType: 'json'
+    }).done(function(data){
+        currentResult=data.results;
+        //change patient name to selected patient
+        $('#by-name').val(null);
+        var newOption = "<option value='" +currentResult.pid+ "' selected>"+currentResult.lname + ', ' + currentResult.fname+"</option>";
+        $('#by-name').append(newOption);
+    })
+});
+
+//get all the data of selected patient
+$('#by-name').on('change', function () {
+    $.ajax({
+        url: 'multi_patients_finder_ajax.php',
+        data:{
+            type:'patient-by-id',
+            search:$('#by-name').val()
+        },
+        dataType: 'json'
+    }).done(function(data){
+        currentResult=data.results;
+        //change patient pubpid to selected patient
+        $('#by-id').val(null);
+        var newOption = "<option value='" +currentResult.pid+ "' selected>"+ currentResult.pubpid +"</option>";
+        $('#by-id').append(newOption);
+    })
+});
+
+//add new patient to list
+$('#add-to-list').on('click', function (e) {
+    e.preventDefault();
+
+    if($('#by-name').val() == '')return;
+
+    if(patientsList.length === 0){
+        $('#results-table').show();
+    }
+    
+    // return if patient already exist in the list
+    var exist
+    $.each(patientsList, function (key, patient) {
+        if (patient.pid == currentResult.pid) exist = true;
+    })
+    if(exist)return;
+
+    
+    // add to array
+    patientsList.push(currentResult);
+    var lastIndex = patientsList.length-1;
+
+    $('#searchResults').append('<tr id="row'+lastIndex+'">' +
+        '<td>'+ currentResult.lname + ', ' + currentResult.fname + '</td>' +
+        '<td>' + currentResult.phone_home + '</td>' +
+        '<td>' + currentResult.ss + '</td>' +
+        '<td>' + currentResult.DOB + '</td>' +
+        '<td>' + currentResult.pubpid + '</td>' +
+        '<td><i class="fa fa-remove remove-patient" onclick="removePatient('+lastIndex+')"></i></td>' +
+    '<tr>');
+
+});
+
+// remove patient from list
+function removePatient(index) {
+    patientsList.splice(index,1);
+    $('#row'+index).remove();
+}
+
+//send array of patients to function 'setMultiPatients' of the opener
+function selPatients() {
+    if (opener.closed || ! opener.setMultiPatients)
+        alert("<?php echo htmlspecialchars(xl('The destination form was closed; I cannot act on your selection.'), ENT_QUOTES); ?>");
```

Comment:
remove htmlspecialchar and xl here
and use xls() instead

Reply:
Done